### ISSUE TITLE
Develop unit test cases for validating URLs

### ISSUE THREAD

**Issue by Prathitha** [shah/uniform-resource#2]
Develop unit test cases for validating URLs. This includes all type of validations.

**Comment by shah** [comment]
@Prathitha as a good learning exercise please either (a) see if you're willing to send me a PR to add these tests or (b) assign a junior dev to do the same.

**Comment by Prathitha** [comment]
Please find the attached file for different types of urls and their expectations.
[uniform resource.xlsx](https://github.com/shah/uniform-resource/files/5082916/uniform.resource.xlsx)

**Comment by Prathitha** [comment]
Please find attached the file for different types of URLs and their expectations. Please advise if this is the correct way. 
Thanks,
[Urls and Expectations.xlsx](https://github.com/shah/uniform-resource/files/5082941/Urls.and.Expectations.xlsx)

**Comment by shah** [comment]
Yes @Prathitha this is great - please LMK if you all will be updating the code and sending PRs or if I should do it.

**Comment by Prathitha** [comment]
Thanks @shah. I will be able to provide you test urls and expectations rather than updating the code.
